### PR TITLE
Add docker image to build protobufs

### PR DESCRIPTION
**Pull request by bogdandrutu**
Forked from https://github.com/jaegertracing/docker-protobuf

### CODE REVIEW

File: .github/workflows/protobuf-dockerimage.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
+        function tag_and_push {
+          docker tag build-protobuf "otel/build-protobuf:${1}" && docker push "otel/build-protobuf:${1}"
+        }
+        if [[ "${GITHUB_REF}" == "refs/heads/master" ]]; then
+          tag_and_push "latest"
+        elif [[ "${GITHUB_REF}" =~ refs/tags/v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
+            TAG="${GITHUB_REF#"refs/tags/v"}"
+            tag_and_push "${TAG}"
+            tag_and_push "latest"
```
Comment (SergeyKanzhelev) [inline .github/workflows/protobuf-dockerimage.yml:32]: what if it's a tag for an old release (like a hotfix)? Should it still push for latest?

File: protobuf/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+docker run --rm -v<some-path>:<some-path> -w<some-path> otel/build-protobuf [OPTION] PROTO_FILES
+```
+
+For help try:
+
+```bash
+docker run --rm otel/build-protobuf --help
+```
+
+### To generate language specific code
```
Comment (SergeyKanzhelev) [inline protobuf/README.md:42]: btw, do we keep these in proto files?

https://github.com/open-telemetry/opentelemetry-proto/blob/dfa8c325321d55cd9e27fe6e2c7bf927bb4d9ca3/opentelemetry/proto/collector/logs/v1/logs_service.proto#L24-L27

File: .github/workflows/protobuf-dockerimage.yml (hunk 1 above)
Comment (bogdandrutu) [reply .github/workflows/protobuf-dockerimage.yml:32]: Done.

File: protobuf/README.md (hunk 2 above)
Comment (bogdandrutu) [reply protobuf/README.md:42]: Those are proto options, here I am talking about external proto files dependencies.

**Pull request merged**